diff --git a/src/Algebra/Graph.hs b/src/Algebra/Graph.hs
index d6a8d19..dccebc4 100644
--- a/src/Algebra/Graph.hs
+++ b/src/Algebra/Graph.hs
@@ -1029,7 +1029,7 @@ pairs as@(x:xs) = zip as (xs ++ [x])
 -- n > 0 ==> 'edgeCount'   (deBruijn n xs) == ('length' $ 'Data.List.nub' xs)^(n + 1)
 -- @
 deBruijn :: Int -> [a] -> Graph [a]
-deBruijn 0   _        = edge [] []
+deBruijn 0   _        = vertex []
 deBruijn len alphabet = skeleton >>= expand
   where
     overlaps = mapM (const alphabet) [2..len]
