diff --git a/src/Algebra/Graph/Label.hs b/src/Algebra/Graph/Label.hs
index ab1ceac..f1d617b 100644
--- a/src/Algebra/Graph/Label.hs
+++ b/src/Algebra/Graph/Label.hs
@@ -323,10 +323,10 @@ noMinimum :: Minimum a
 noMinimum = Minimum Infinite
 
 instance Ord a => Semigroup (Minimum a) where
-    (<>) = min
+    (<>) = liftA2 min
 
 instance (Monoid a, Ord a) => Monoid (Minimum a) where
-    mempty = noMinimum
+    mempty = pure mempty
 
 instance (Monoid a, Ord a) => Semiring (Minimum a) where
     one   = pure mempty
